File: python/easy/count_primes.py
class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        import math
        if n <= 1:
            return 0
        primes = [True]*n
        j = 0
        for i in xrange(2, int(math.sqrt(n))+1):
            j = i * 2
            while j  < n:
                primes[j] = False
                j += i
        return primes.count(True)-2
        

File: cpp/easy/count_primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        if(n <= 1)
            return 0;
        vector<bool> primes(n,true);
        primes[0] = false;
        primes[1] = false;
        int count = 0;
        for(int i = 2; i < sqrt(n); ++i)
        {
            int j = i*i;
            while(j < n)
            {
                primes[j] = false;
                j += i;
            }
        }
        
        for(int i = 2; i < primes.size(); ++i)
        {
            if(primes[i])
                ++count;
        }
        return count;
    }
};


Conclusion: an easy algorithm issue, no any other comments.

Time Complexity: O(n)
Space Complexity: O(1)
